### DISCUSSION TITLE
Integration with MDX using Custom Components

### DISCUSSION THREAD

**Discussion by iamhectorsosa** [withastro/roadmap#372] [Proposal]
Hello, Astro community:

I'm testing Astro's official integration with MDX (and Astro in general). Roughly speaking, I am using the following structure:

```graphql
├── pages
│   └── index.mdx     # <-- MDX with custom components
├── layouts
│   └── Base.astro    # <-- Layout that is being used for MDX pages
```
I am reading through the docs and it says that once we create a custom component we can then import it and export it as follows:

```javascript
/* src/pages/posts/post-1.mdx */
import Blockquote from '../components/Blockquote.astro';
export const components = { blockquote: Blockquote };
```

The approach of importing and exporting custom component on each MDX page doesn't seem very scalable. Is there a way to do this at the layout level?

Thanks in advance! Any ideas would be appreciated!

This is the source: [Integrations Guide MDX Custom Components](https://docs.astro.build/en/guides/integrations-guide/mdx/#custom-components)

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
